xdsl/dialects/csl.py

```diff
@@ -1487,32 +1487,66 @@ class ParamOp(IRDLOperation):
     res = result_def(T)
 
 
+@irdl_op_definition
+class SignednessCastOp(IRDLOperation):
+    """
+    Cast that throws away signedness attributes
+    """
+
+    name = "csl.mlir.signedness_cast"
+
+    inp = operand_def(IntegerType)
+
+    result = result_def(IntegerType)
+
+    assembly_format = "$inp attr-dict `:` type($inp) `to` type($result)"
+
+    def verify_(self) -> None:
+        assert isinstance(self.inp.type, IntegerType)
+        assert isinstance(self.result.type, IntegerType)
```

Comment:
Are these `assert`s needed for the sake of `pyright`?

Reply:
They are enforced by `irdl`, so we don't need to manually verify